File: SDKs/amazonaws.com/cloudhsmv2/2017-04-28/go/pkg/models/operations/modifybackupattributes.go
package operations

import (
"openapi/pkg/models/shared")


type ModifyBackupAttributesXAmzTargetEnum string

const (
    ModifyBackupAttributesXAmzTargetEnumBaldrAPIServiceModifyBackupAttributes ModifyBackupAttributesXAmzTargetEnum = "BaldrApiService.ModifyBackupAttributes"
)


type ModifyBackupAttributesHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget ModifyBackupAttributesXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type ModifyBackupAttributesRequest struct {
    Headers ModifyBackupAttributesHeaders 
    Request shared.ModifyBackupAttributesRequest `request:"mediaType=application/json"`
    
}

type ModifyBackupAttributesResponse struct {
    CloudHsmAccessDeniedException *interface{} 
    CloudHsmInternalFailureException *interface{} 
    CloudHsmInvalidRequestException *interface{} 
    CloudHsmResourceNotFoundException *interface{} 
    CloudHsmServiceException *interface{} 
    ContentType string 
    ModifyBackupAttributesResponse *shared.ModifyBackupAttributesResponse 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/cloudhsmv2/2017-04-28/typescript/src/sdk/models/operations/modifybackupattributes.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum ModifyBackupAttributesXAmzTargetEnum {
    BaldrApiServiceModifyBackupAttributes = "BaldrApiService.ModifyBackupAttributes"
}


export class ModifyBackupAttributesHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: ModifyBackupAttributesXAmzTargetEnum;
}


export class ModifyBackupAttributesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: ModifyBackupAttributesHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ModifyBackupAttributesRequest;
}


export class ModifyBackupAttributesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  cloudHsmAccessDeniedException?: any;

  @SpeakeasyMetadata()
  cloudHsmInternalFailureException?: any;

  @SpeakeasyMetadata()
  cloudHsmInvalidRequestException?: any;

  @SpeakeasyMetadata()
  cloudHsmResourceNotFoundException?: any;

  @SpeakeasyMetadata()
  cloudHsmServiceException?: any;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  modifyBackupAttributesResponse?: shared.ModifyBackupAttributesResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}
